test/test_assembleP3.jl

```diff
@@ -0,0 +1,104 @@
+        fill!(Ke, 0)
+        reinit!(cellvalues, cell)
+        for q_point in 1:getnquadpoints(cellvalues)
+            dΩ = getdetJdV(cellvalues, q_point)
+            for i in 1:n_basefuncs
+                v  = shape_value(cellvalues, q_point, i)
+                ∇v = shape_gradient(cellvalues, q_point, i)
+                for j in 1:n_basefuncs
+                    ∇u = shape_gradient(cellvalues, q_point, j)
+                    Ke[i, j] += (∇v ⋅ ∇u) * dΩ
+                end
+            end
+        end
+        assemble!(assembler, celldofs(cell), Ke)
+    end
+    return K
+end
+
+function doassemble_M_ferrite!(M::SparseMatrixCSC, 
+                       cellvalues::CellScalarValues{dim}, dh::DofHandler;
+                       ρ::Union{Function, Nothing} = nothing) where {dim}
+    n_basefuncs = getnbasefunctions(cellvalues)
+    Me = zeros(n_basefuncs, n_basefuncs)
+    assembler = start_assemble(M)
+    valρ = 1.
+    @inbounds for cell in CellIterator(dh)
+        fill!(Me, 0)
+        reinit!(cellvalues, cell)
+        for q_point in 1:getnquadpoints(cellvalues)
+            dΩ = getdetJdV(cellvalues, q_point)
+            if ρ != nothing 
+                pq = spatial_coordinate(cellvalues, q_point, 
+                                        getcoordinates(cell))
+                valρ = ρ(pq)
+            end
+            for i in 1:n_basefuncs
+                v  = shape_value(cellvalues, q_point, i) 
+                for j in 1:n_basefuncs
+                    u = shape_value(cellvalues, q_point, j)                   
+                    Me[i, j] += (v * u) * valρ * dΩ
+                end
+            end
+        end
+        assemble!(assembler, celldofs(cell), Me)
+    end
+    return M
+end
+function test_H1Pk(; degfem::Int64 = 2, degcurve::Int64 = 1, degquad::Int64 = 6,
+                     dim::Int64 = 2)
+    typecellq = degcurve == 2 ? QuadraticTriangle : Ferrite.Triangle
+    grid = generate_grid(typecellq, (30, 30))
+    ip = Ferrite.Lagrange{dim, RefTetrahedron, degfem}()
+    qr = Ferrite.QuadratureRule{dim, RefTetrahedron}(:legendre, degquad)
+    ip_geom = Ferrite.Lagrange{dim, RefTetrahedron, degcurve}() 
+    cellvalues = Ferrite.CellScalarValues(qr, ip, ip_geom)
+    dh = Ferrite.DofHandler(grid)
+    Ferrite.push!(dh, :u, 1, ip)
+    Ferrite.close!(dh)
+
+    K = create_sparsity_pattern(dh)
+    doassemble_K_ferrite!(K, cellvalues, dh)
+    M = create_sparsity_pattern(dh)
+    doassemble_M_ferrite!(M, cellvalues, dh)
+
+    # compute eigenvalues
+    λ, U = eigs(K, M, nev = 10, which = :LM, tol = 0., maxiter = 30_000, 
+                sigma = rand() / 1e4) 
+    # display
+    error = maximum(abs.([0, 1/4, 1/4, 1/2, 1, 1, 5/4, 5/4] * pi ^ 2 - λ[1:8]))
+    nbdof = size(K, 1)
+    @show real.(λ)
+    @show nbdof
+    @show error
+    @test error < 0.1
+    nbdof = size(K, 1)
+
+    return 
+end
+#-------------------------------------------------------------------------------
+function test_ip(ip)
```

Comment:
I believe we have a test function like this in the `test_interpolations.jl` file, perhaps just test your new interpolations in that file?